enhancements/local-storage/automatic-partition.md

```diff
@@ -0,0 +1,244 @@
+	Spec              LocalVolumeSetSpec   `json:"spec"`
+	Status            LocalVolumeSetStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeSetSpec struct {
+  // Nodes on which the automatic detection policies must run.
+  // +optional
+  NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+  // StorageClassName to use for set of matched devices
+  StorageClassName string `json:"storageClassName"`
+  // MinDeviceCount is the minumum number of devices that needs to be detected per node.
+  // If the match devices are less than the minCount specified then no PVs will be created.
+  // +optional
+  MinDeviceCount *int32 `json:"minDeviceCount"`
+  // Maximum number of Devices that needs to be detected per node.
+  // +optional
+  MaxDeviceCount *int32 `json:"maxDeviceCount"`
+  // VolumeMode determines whether the PV created is Block or Filesystem. By default it will
+  // be block
+  // + optional
+  VolumeMode PersistentVolumeMode `json:"volumeMode,omitempty"`
+  // FSType type to create when volumeMode is Filesystem
+  // +optional
+  FSType string `json:"fsType,omitempty"`
+  // If specified, a list of tolerations to pass to the discovery daemons.
+  // +optional
+  Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+  // DeviceInclusionSpec is the filtration rule for including a device in the device discovery
+  // +optional
+  DeviceInclusionSpec *DeviceInclusionSpec `json:"deviceInclusionSpec"`
+  
+  // PartitioningSpec is the partitioning policy according to which the selected devices will be partitioned.
+  PartitioningSpec PartitoiningSpec `json:"partitioningSpec"`
+}
+
+type PartitoiningSpec struct {
+  // PartSize determines the exact size of each partition.
+  PartSize *resource.Quantity `json:"partSize"`
+}
+
+type LocalVolumeSetStatus struct {
+  // Conditions is a list of conditions and their status.
+  Conditions []operatorv1.OperatorCondition `json:"conditions,omitempty"`
+  
+  // TotalProvisionedDeviceCount is the count of the total devices over which the PVs has been provisioned
+  TotalProvisionedDeviceCount *int32 `json:"totalProvisionedDeviceCount,omitempty"`
+  
+  // observedGeneration is the last generation change the operator has dealt with
+  // +optional
+  ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+  
+  // TotalProvisionedPartitionCount is the count of the total partitions over which the PVs has been provisioned
+  // This will be diffrent for TotalProvisionedDeviceCount as TotalProvisionedDeviceCount is
+  //the count of the actual devices
+  TotalProvisionedPartitionCount *int32 `json:"totalProvisionedPartitionCount,omitempty"`
+  
+  // PartitioningStatus will track the status of partitioning
+  // Valid values are
+  // nil : if there is no partitioningSpec
+  // partitoining : if the partitioning is in progress
+  // completed: when the partitioning of devices is completed
+  PartitioningStatus string `json:"partitioningStatus"`
```

Comment:
We shouldn't need a special partitioning status. It's just an overall status for reconciling the LocalVolumeSet, whether devices or partitions.